contracts/core/LensHub.sol

```diff
@@ -531,7 +531,8 @@ contract LensHub is ILensHub, LensNFTBase, VersionedInitializable, LensMultiStat
         whenNotPaused
     {
         bytes32[] memory dataHashes = new bytes32[](vars.datas.length);
-        for (uint256 i = 0; i < vars.datas.length; ) {
+        uint256 dataLength = vars.datas.length;
+        for (uint256 i = 0; i < dataLength; ) {
```

Comment:
Do it changes anything if we instead do this?

```solidity
        uint256 dataLength = vars.datas.length;
        bytes32[] memory dataHashes = new bytes32[](dataLength);
        for (uint256 i = 0; i < dataLength; ) {
```

Reply:
maybe missed it, let's see if it works!

Reply:
It went well, we saved an extra 3k of gas, from 5146288 to 5143232. Will push the change!